src/main/java/tc/oc/pgm/menu/InventoryMenu.java

```diff
@@ -0,0 +1,151 @@
+package tc.oc.pgm.menu;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.Sets;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.bukkit.Bukkit;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+import tc.oc.component.render.ComponentRenderers;
+import tc.oc.component.types.PersonalizedTranslatable;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.util.StringUtils;
+
+public abstract class InventoryMenu {
+
+  protected static final int ROW_WIDTH = 9; // Number of columns per row
+  protected static final int MAX_ROWS = 6; // Max allowed row size
+
+  private final Set<MatchPlayer> viewing =
+      Sets.newHashSet(); // Set of players who are viewing the gui
```

Comment:
This doesn't seem to get cleaned up anywhere? How about using `WeakHashMap` ?

Reply:
Hey @Electroid,
The viewing map does get cleaned up via the ObserverToolMatchModule, though that is only for that implementation. I switched it over to a `WeakHashMap` as suggested, to accommodate for when future implementations are added. 
Thanks!